sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatClientBuilder.java

```diff
@@ -203,7 +219,7 @@ private HttpPipeline createHttpPipeline(HttpClient httpClient,
 
     private void applyRequiredPolicies(List<HttpPipelinePolicy> policies) {
         policies.add(getUserAgentPolicy());
-        policies.add(new RetryPolicy());
+        policies.add(this.retryPolicy == null ? new RetryPolicy() : this.retryPolicy);
```

Comment:
Just a question here: Is this ever possible for `this.retryPolicy == null`?

Reply:
Yup, if the retryPolicy is not set, then it will be null